packages/babel-traverse/src/visitors.ts

```diff
@@ -104,13 +104,9 @@ function explode$1<S>(visitor: Visitor<S>): ExplodedVisitor<S> {
     const types = virtualTypes[nodeType];
     if (types !== null) {
       for (const type of types) {
-        // merge the visitor if necessary or just put it back in
-        if (visitor[type]) {
-          mergePair(visitor[type], fns);
-        } else {
-          // @ts-expect-error Expression produces too complex union
-          visitor[type] = fns;
-        }
```

Comment:
This is a bug fix, previously it shallow copied `fns` to multiple places, but it just happened not to be modified. After we added the definition of `BlockScoped` it was modified later, causing the visitor to be wrong. Here we use `mergePair` to do a deep clone.